front/src/modules/timesStops/TimesStopsInput.tsx

```diff
@@ -56,13 +55,31 @@ const TimesStopsinput = ({ allWaypoints, startTime, pathSteps }: TimesStopsInput
       if ('uic' in step) {
```

Comment:
For readability, maybe you can use something like that, so the extracted functions can be used in `findIndex`;

```
const isMatchingUICStep = (step: PathStep) =>
      'uic' in step &&
      step.uic === rowData.uic &&
      step.ch === rowData.ch &&
      step.name === rowData.name;

    const isMatchingTrackStep = (step: PathStep) =>
      'track' in step &&
      step.track === rowData.track &&
      step.positionOnPath === rowData.positionOnPath &&
      step.offset === rowData.offsetOnTrack &&
      step.id === rowData.opId;

    const index = pathSteps.findIndex(
      (step) => isMatchingUICStep(step) || isMatchingTrackStep(step)
    );
```

Reply:
You're right. It's done.
I also rename `removeVia` in `clearPathStep` because I find this name more appropriate.